Automattic/wp-calypso
# Improve guided tours lazy loading

Guided Tours were already being lazily loaded, so that the significant amount of code they pull in wasn't part of the main bundles.

However, this was still eventually being loaded for every single user, regardless of whether they were to be shown a guided tour.

This PR separates display logic from data logic, so that it's possible to avoid loading the visual components altogether if no guided tour is to be displayed.

For users without an active guided tour, this PR saves about 19KB compressed JS.

#### Changes proposed in this Pull Request

* Create a directory for each tour
* Separate metadata for each tour into its own file
* Move main Guided Tour logic to a new component, and rewrite the main component to lazily load that only if needed.


#### Testing instructions

To validate correctness:

* Open an account with active guided tours
* Ensure the guided tour functionality still works
* Open an account with no active guided tours
* Ensure there are no errors

To validate performance gains:

* In an account with no active guided tours, open https://hash-0851bf61c6723321876eb66ef8e634e5c322d5c1.calypso.live/ (branch point before this PR)
* Open DevTools, switch to the Network tab and filter by JS
* In another tab and in the same account, open https://calypso.live/?branch=update/improve-guided-tours-lazy-loading (this branch)
* Open DevTools, switch to the Network tab and filter by JS
* Verify that there is less JS being loaded in the second case than in the first.
